src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/SendOrderContestedAboutToSubmitHandler.java

```diff
@@ -170,12 +173,19 @@ private void shareAndSendGeneralOrderWithSelectedParties(FinremCaseDetails caseD
         log.info("Share selected 'GeneralOrder' With selected parties for caseId {}", caseId);
 
         FinremCaseData caseData = caseDetails.getData();
-        CaseDocument generalOrder = caseData.getGeneralOrderWrapper().getGeneralOrderLatestDocument();
 
-        if (generalOrderService.isSelectedOrderMatches(selectedOrders, generalOrder)) {
-            sendOrderPartyDocumentList.forEach(
-                handler -> handler.setUpOrderDocumentsOnCase(caseDetails, partyList, List.of(generalOrder)));
-            printOrderCollection.add(addToPrintOrderCollection(generalOrder));
+        List<ContestedGeneralOrderCollection> generalOrders = caseData.getGeneralOrderWrapper().getGeneralOrders();
```

Comment:
We have each parties handler, I would recon to move this logic into abstract class in SendOrderPartyDocumentHandler

Reply:
this check needed in case Case do not have general order - we don't want to call party handler

Reply:
Ah ok, here we can use collectionUtils to check null and empty()